### PR TITLE
Initialize Drivers when they exists!

### PR DESCRIPTION
**Pull request by iiSaLMaN**
Initialize Drivers when their devices/controllers exist so anyone can boot their opearting system on any machine with any device :-) ,

~Currently i'm working on AHCI driver #763~

### CODE REVIEW

File: source/Cosmos.HAL2/Global.cs [hunk 1]
```diff
@@ -46,15 +46,24 @@ static public void Init(TextScreenBase textScreen)
...
+        InitAta(Ata.ControllerIdEnum.Secondary, Ata.BusPositionEnum.Master);
+        //InitAta(BlockDevice.Ata.ControllerIdEnum.Secondary, BlockDevice.Ata.BusPositionEnum.Slave);
+      }
+      else if (PCI.GetDeviceClass(0x01, 0x06) != null)
+      {
+        mDebugger.Send("AHCI isn't supported yet");
+        Console.Write("AHCI Controller not supported yet");
+        Console.Write("Booting without ATA Initialization");
+      }
+      else if (PCI.GetDeviceClass(0x01, 0x01) == null)
```
Comment (zarlo) [inline source/Cosmos.HAL2/Global.cs]: this can just be an else

File: source/Cosmos.HAL2/Global.cs [hunk 2]
```diff
@@ -46,15 +46,24 @@ static public void Init(TextScreenBase textScreen)
...
-      mDebugger.Send("ATA Primary Master");
-      InitAta(Ata.ControllerIdEnum.Primary, Ata.BusPositionEnum.Master);
-
-      //TODO Need to change code to detect if ATA controllers are present or not. How to do this? via PCI enum?
-      // They do show up in PCI space as well as the fixed space.
-      // Or is it always here, and was our compiler stack corruption issue?
-      mDebugger.Send("ATA Secondary Master");
-      InitAta(Ata.ControllerIdEnum.Secondary, Ata.BusPositionEnum.Master);
-      //InitAta(BlockDevice.Ata.ControllerIdEnum.Secondary, BlockDevice.Ata.BusPositionEnum.Slave);
+      if (PCI.GetDeviceClass(0x01, 0x01) != null)
```
Comment (jp2masa) [inline source/Cosmos.HAL2/Global.cs]: Please add a comment about the meaning of these numbers and the ones on line 57 too.

**Comment by KingLuigi4932** [comment]
Done! @jp2masa

File: source/Cosmos.HAL2/Global.cs [hunk 3]
```diff
@@ -45,16 +45,45 @@ static public void Init(TextScreenBase textScreen)
...
+      // Subclass 0x04 = Subclass code of RAID Controller
+      // Subclass 0x06 = Subclass code of AHCI Controller
+      if (PCI.GetDeviceClass(0x01, 0x00) != null)
+      {
+        mDebugger.Send("SCSI isn't supported yet");
+        Console.WriteLine("SCSI Controller not supported yet");
+        Console.WriteLine("Booting without ATA Initialization");
+        Console.WriteLine("FAT cannot be used while ATA isn't initialized"); 
+      }
+      else if (PCI.GetDeviceClass(0x01, 0x01) != null)
```
Comment (jp2masa) [inline source/Cosmos.HAL2/Global.cs]: I think that this should be the first check. If ATA is detected, we don't really care about the others.

**Comment by iiSaLMaN** [comment]
Closing this PR as it has been abandoned for over a year and fork has been lost, Sorry guys >_<

**Pull request closed without merging**